sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -247,11 +247,18 @@ public final BlobInputStream openInputStream(BlobRange range, BlobRequestConditi
      */
     public BlobInputStream openInputStream(BlobInputStreamOptions options) {
         options = options == null ? new BlobInputStreamOptions() : options;
+
+        BlobProperties properties = getProperties();
+
         BlobRange range = options.getRange() == null ? new BlobRange(0) : options.getRange();
         int chunkSize = options.getBlockSize() == null ? 4 * Constants.MB : options.getBlockSize();
 
+        BlobRequestConditions requestConditions = options.getRequestConditions() == null
+            ? new BlobRequestConditions() : options.getRequestConditions();
+        requestConditions.setIfMatch(properties.getETag());
```

Comment:
I think we should also check that there aren't existing etag conditions. If they pass in an old etag that they want to target, and we just got a new one possibly after some updates, we might not be reading the version they requested.

Reply:
Yeah makes sense

Reply:
Fixed.